.github/actions/setup-oblt-cli/action.yml

```diff
@@ -42,3 +42,13 @@ runs:
     - name: Export env
       run: echo "ELASTIC_APM_ENVIRONMENT=ci" >> $GITHUB_ENV
       shell: bash
+    - name: Create workloadIdentityPool provider ID
+      id: generate-workload-identity-pool-provider-id
+      run: |
+        HASH=$(echo "${GITHUB_WORKFLOW_REF}" | awk -F@ '{printf $1}' | sha256sum | cut -c1-23)
+        echo "workload_identity_provider_id=workflow-${HASH}" >> "${GITHUB_OUTPUT}"
+      shell: bash
+    - uses: 'google-github-actions/auth@v2'
+      with:
+        project_id: 'elastic-observability'
+        workload_identity_provider: 'projects/8560181848/locations/global/workloadIdentityPools/github/providers/${{ steps.generate-workload-identity-pool-provider-id.outputs.workload_identity_provider_id }}'
```

Comment:
Do you think we can provide a GitHub composite action in charge of doing this? So any consumers can benefit from?

Or maybe that's something to be implicitly called by the `setup-oblt-cli`. What do you think?

Reply:
I know this is for testing purposes :)

Reply:
I was actually thinking of creating a wrapper google auth action with this code.

which can be used at the beginning of every job, where it's needed.

Reply:
Another Idea I had.. was to save the workload identity id as a GitHub secret.

This way.. only workflows with access to secret could authenticate.

Of course, this assumes that the ID is obscured and not backtrackable—security by obscurity.

The algorithm to create the ID would need to change. Since this PR is already public 😅

Reply:
We can start with some simple.

I think the wrapper is neat, and maybe `setup-oblt-cli` can use it if no vault inputs are passed?

Reply:
I don't think we should use it by default.

I think it's a fair use case to have other resources than just oblt-cli and authenticating.

So, IMO, authenticating should happen outside of the setup-oblt-cli step.

However, maybe we can have a convenience flag to do the authentication?

Reply:
I think `oblt-cli` is quite an opinionated tool that requires Google authentication or vault access, regardless. IMO, consumers should not care about this, for the shake of simplicity.

I'd expect a simple way of using it like:

```yaml
jobs:
  run-oblt-cli-create-serverless:
    runs-on: ubuntu-latest
    steps:
      - uses: ./.github/actions/oblt-cli-create-serverless
        with:
          token: ${{ secrets.MY_TOKEN }}
          target: 'qa'
          cluster-name-prefix: 'testgithubaction'
```

rather than

```yaml
jobs:
  run-oblt-cli-create-serverless:
    runs-on: ubuntu-latest
    steps:
      - uses: ./.github/actions/setup-auth
         with:
            ....
      - uses: ./.github/actions/oblt-cli-create-serverless
        with:
          token: ${{ secrets.MY_TOKEN }}
          target: 'qa'
          cluster-name-prefix: 'testgithubaction'
```

> However, maybe we can have a convenience flag to do the authentication?

I don't envision we will use it with default `false`, therefore we will need to set it to true explicitly.

Do you envision a different use case?

Reply:
So, my thinking is.

The more we use keyless, the more resources we will need.

Hence, there will be workflows where we use both oblt-cli, or terraform, or gcloud.

Then the question arises, when should we do the auth step?

I think it can lead to confusion if oblt-cli is authenticating by default.

It could break existing authentication attempts in previous steps.

I think it could be a similar  reason, why these actions are separated e.g. `google-github-actions/setup-gcloud` and ``google-github-actions/auth`

Reply:
Separating this intentionally makes it implicit, and having docs that show the requirement for this.. could be enough.

Reply:
Generally, I agree with you.. making authenticate by default is the better user experience.


But I think a strong requirement for this is.. that we don't mess up existing authentications to GCP.

Reply:
I will dig a bit more into this.

Reply:
Unfortunately, I was not able to find a way to authenticate in an isolated way.